### PR TITLE
MGMT-15346: Adds pkg/ignition unit-tests

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

File: pkg/ignition/ignition.go [hunk 1]
```diff
@@ -4,52 +4,59 @@ import (
...
 
-	ignitionConfigPrevVersion "github.com/coreos/ignition/v2/config/v3_1"
 	ignitionConfig "github.com/coreos/ignition/v2/config/v3_2"
-	"github.com/coreos/ignition/v2/config/v3_2/translate"
 	"github.com/coreos/ignition/v2/config/v3_2/types"
 	"github.com/coreos/ignition/v2/config/validate"
 	"github.com/pkg/errors"
 )
 
+type OSInterface interface {
```
Comment (danielerez) [inline pkg/ignition/ignition.go:13]: again? :) so we have an interface for each package for now? :/

Comment (nmagnezi) [reply pkg/ignition/ignition.go:13]: Will unify it all in a dedicated PR.

Comment (nmagnezi) [reply pkg/ignition/ignition.go]: ok. done. it is now using the `fileutil.OSInterface`.

File: pkg/ignition/ignition_test.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+		},
+		"storage": {
+		  "files": []
+		}
+	  }`
+)
+
+type FakeOS struct{}
+
+func (FakeOS) MkdirTemp(dir, prefix string) (string, error) {
```
Comment (danielerez) [inline pkg/ignition/ignition_test.go:39]: nit for later: looks mostly similar to the struct in release_test. Maybe we can try to extract it later on. So we could override only the needed funcs.

**Review by danielerez — COMMENTED** [review]
partially reviewed

**Pull request merged**